library/modules/EventManager.cpp

```diff
@@ -597,57 +634,77 @@ static void manageJobCompletedEvent(color_ostream& out) {
     }
 #endif
 
-    for (auto &prevJob : prevJobs) {
-        //if it happened within a tick, must have been cancelled by the user or a plugin: not completed
-        if ( tick1 <= tick0 )
-            continue;
-
-        if ( nowJobs.find(prevJob.first) != nowJobs.end() ) {
-            //could have just finished if it's a repeat job
-            df::job& job0 = *prevJob.second;
-            if ( !job0.flags.bits.repeat )
-                continue;
-            df::job& job1 = *nowJobs[prevJob.first];
-            if ( job0.completion_timer != 0 )
-                continue;
-            if ( job1.completion_timer != -1 )
-                continue;
-
-            //still false positive if cancelled at EXACTLY the right time, but experiments show this doesn't happen
-            for (auto &[_,handle] : copy) {
-                DEBUG(log,out).print("calling handler for repeated job completed event\n");
-                run_handler(out, EventType::JOB_COMPLETED, handle, (void*) &job0);
+    /*
+     * if it happened within a tick, must have been cancelled by the user or a plugin: not completed
+     * note observation: delta 0 tick events happens normally on load, save, paused (whether by user or game).
+     * Does it make sense to check this here? This should be handled by manageEvents().
+     */
+    if (tick1 >= tick0) {
```

Comment:
shouldn't this be `tick1 > tick0`? but you're also right that this should already be gated by the check in `manageEvents`: `if ( tick >= eventLastTick[a] && tick - eventLastTick[a] < eventFrequency )`

so this check can probably be removed

Reply:
I'd rather remove it. I changed it to tick1 >= tick0 so that it'd work for 0 tick resolution.